### ISSUE TITLE
Need a feature to stop un-necessary plugins and extensions. 

### ISSUE THREAD

**Issue by a4arpon** [zed-industries/zed#12881]
### Check for existing issues

- [X] Completed

### Describe the feature

I hate vs-code because of it's performance issue. That's why I am willing to start development with Zed. 

But the problem is feature's I am not using still take memory by zed. Like I am using Biome Linter for all of my projects now. But Zed ships a Prettier and EsLint package by default. And they are consuming 150MB rams total. But it's not the only thing. I am not using the co-pilot extension but it taking 143 or 135 mbs of Ram every time. And I am sure that I will never co-pilot in future. But what is the point to ship it dy default. Okay this is good you shipped but after disabling it's running at the back-ground. So what is the point to disable it. 

please make a way that we can disable un-necessary features for us completely.   

### If applicable, add mockups / screenshots to help present your vision of the feature

![image](https://github.com/zed-industries/zed/assets/70208852/e971152c-c068-48f6-b006-f87c4679fe6d)

Zed consuming ram for Co-Pilot when I disabled it already.

![image](https://github.com/zed-industries/zed/assets/70208852/850f14b9-d942-4024-bfcf-c5edc76cb8dd)
I am using BiomeJS for linting but Zed takes ram for EsLint.

![image](https://github.com/zed-industries/zed/assets/70208852/11283b57-94b9-417d-8a2e-65afe22233ff)
Some features consuming memory while I am not using them. Zed needs a feature to completely disabled them

**Comment by chungweileong94** [comment]
Yeah, having an option to disable the extension would be nice. As of now, most of the things that you mentioned can be disabled via settings.

For copilot-related features, you can do:
```json
"features": { "inline_completion_provider": "none" },
"assistant": { "version": "1", "enabled": false },
```
Other things like Eslint & Prettier, they are mostly LSP, which you can turn them off by doing:
```json
  "language_servers": [ "!prettier", "!eslint", "biome", "typescript-language-server", "!vtsls", "..." ]
```
> Do note that you need to specify `"typescript-language-server", "!vtsls"` starting from v0.139, else both LSP will be utilize for `.ts, .js, .tsx`

**Comment by maxdeviant** [comment]
> Edit: I think copilot feature process will still running in the background even if you disabled it.

What happens if you restart Zed with it disabled?

I'm testing it myself, and—at least on macOS—when I have `"inline_completion_provider": "none"` in my settings I do not see the Copilot process running.

**Comment by chungweileong94** [comment]
> I'm testing it myself, and—at least on macOS—when I have `"inline_completion_provider": "none"` in my settings I do not see the Copilot process running.

My bad, you are right, I forgot about it being fixed couple of versions ago, wasn't paying too much attention to it since then.

**Comment by notpeter** [comment]
I'm going to go ahead and close this as unactionable.
Zed only spawns LSPs as required for the files you open.

If you would like to disable certain language servers (e.g `biome`, `eslint`) either globally or per language.
See [Configuring Language Servers](https://zed.dev/docs/configuring-languages#choosing-language-servers) in the docs.

As suggested in this similar issue:
- https://github.com/zed-industries/zed/issues/4389

If you don't use eslint you can disable it globally by adding this to your settings:
```
{
  "language_servers": [ "!eslint", "..."]
}
```
